modules/apps/frontend-js/frontend-js-api/types/src/main/resources/META-INF/resources/data-set/index.d.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export interface FDSCellRendererArgs {
+	element: HTMLElement;
+	value: string;
+}
+export interface FDSCellRenderer {
+	(args: FDSCellRendererArgs): void;
```

Comment:
Should the renderer have an enforced return type? From a naming convention, "Renderer" seems weird if it renders via a side effect. I would expect the return type to be either html(string?) or a react component. Or maybe we always expect a string as a return and require a react component to renderToString()

Reply:
Yeah for sure, if we decide to render declaratively (which makes sense to me) we can change this API to return a DOM node/string.

Reply:
Yeah I like the idea of making it declarative and have a return type of string or DOM node. Might be able to get away with just string for now, and then update as needed.